.github/workflows/build-docker.yaml

```diff
@@ -0,0 +1,47 @@
+name: build and push Docker image
+
+on:
+  push:
+    tags:
+      - "release_[0-9]+_test[0-9]+"
+      - "release_[0-9]+"
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: GoogleCloudPlatform/github-actions/setup-gcloud@master
+        with:
+          version: '270.0.0'
+          service_account_email: ${{ secrets.GCP_STG_EMAIL }}
+          service_account_key: ${{ secrets.GCP_STG_PRIVATE_KEY }}
+          export_default_credentials: true
+      - name: Configure GCR Docker auth (stg)
+        run: gcloud auth configure-docker
+      - name: Build docker image
+        run: docker build --tag ml-agents:${GITHUB_SHA} --build-arg SHA=${GITHUB_SHA} .
```

Comment:
Do you want the actual SHA here, or the tag? You can extract the tag from the GITHUB_REF variable, but it will be prefixed by `refs/tags/`, e.g. `refs/tags/release_42`

Reply:
see also: https://docs.github.com/en/free-pro-team@latest/actions/reference/environment-variables

Reply:
Given that the Dockerfile takes a SHA, I think tagging with SHA is reasonable.  Could also add the git ref tag later if needed.